presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -213,13 +230,30 @@ public boolean isEnabled(Session session)
         @Override
         public Result apply(TableScanNode tableScanNode, Captures captures, Context context)
         {
-            if (tableScanNode.getTable().getLayout().isPresent()) {
+            TableHandle tableHandle = tableScanNode.getTable();
+            if (tableHandle.getLayout().isPresent()) {
                 return Result.empty();
             }
 
+            Session session = context.getSession();
+            if (metadata.isPushdownFilterSupported(session, tableHandle)) {
+                PushdownFilterResult pushdownFilterResult = metadata.pushdownFilter(session, tableHandle, TRUE);
```

Comment:
If we push `TRUE` into table scan, we are not expecting taking back any unenforced constraint of `isNone()` case right? Or I might have missed something.

Reply:
Yes, it seems like should be the case.

Reply:
@highker @hellium01 This is existing logic. If the table is empty and the connector has a way to detect this with certainty, it will return `none`. Here is the corresponding snippet of code for the original code path.

```
            TableLayoutResult layout = metadata.getLayout(
                    session,
                    tableHandle,
                    Constraint.alwaysTrue(),
                    Optional.of(tableScanNode.getOutputSymbols().stream()
                            .map(tableScanNode.getAssignments()::get)
                            .collect(toImmutableSet())));

            if (layout.getLayout().getPredicate().isNone()) {
                return Result.ofPlanNode(new ValuesNode(context.getIdAllocator().getNextId(), tableScanNode.getOutputSymbols(), ImmutableList.of()));
            }
```

Reply:
Correct me if I am wrong, but looks to me line 234-245 is sort of doing the same thing as line 248-266, but using the `pushdownFilter` with `TRUE` instead of using `getLayout` with `Constraint.alwaysTrue()`.

I am curious if we can just use the original logic form line 248-266? Since my understanding is `metadata .getLayout` is a more universal method and implemented for every connector. And `pushdownFilter` method is more suitable for `pushPredicateIntoTableScan`.

Or, is the idea here that for a connector supports `pushdownFilter`, this `pushdownFilter` supersedes `getLayout` for consistency purpose?